target/linux/lantiq/files/arch/mips/boot/dts/lantiq/ar10_tplink_archer-d2.dts

```diff
@@ -0,0 +1,268 @@
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+                        partition@0 {
+                                reg = <0x0 0x20000>;
+                                label = "u-boot";
+                                read-only;
+                        };
+
+                        partition@20000 {
+                                reg = <0x20000 0xf90000>;
+                                label = "firmware";
+                        };
+
+                        partition@fb0000 {
+                                reg = <0xfb0000 0x10000>;
+                                label = "radioDECT";
+                                read-only;
+                        };
+
+                        partition@fc0000 {
+                                reg = <0xfc0000 0x10000>;
+                                label = "config";
+                                read-only;
+                        };
+
+                        romfile: partition@fd0000 {
+                                reg = <0xfd0000 0x10000>;
+                                label = "romfile";
+                                read-only;
+                        };
+
+                        partition@fe0000 {
+                                reg = <0xfe0000 0x10000>;
+                                label = "rom";
+                                read-only;
+                        };
+
+                        radio: partition@ff0000 {
+                                reg = <0xff0000 0x10000>;
+                                label = "radio";
+                                read-only;
+                        };
+                };
+        };
+};
+
+&eth0 {
+        pinctrl-names = "default";
+	mtd-mac-address = <&romfile 0xf100>;
+};
+
+&gswip {
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+};
+
+//&gphy0 {
+//	lantiq,gphy-mode = <GPHY_MODE_GE>;
+//};
+//&gphy1 {
+//	lantiq,gphy-mode = <GPHY_MODE_GE>;
+//};
+//&gphy2 {
+//	lantiq,gphy-mode = <GPHY_MODE_GE>;
+//};
+
+
+&gswip_mdio {
+	phy4: ethernet-phy@4 {
+		reg = <0x4>;
+	};
+	phy1: ethernet-phy@1 {
+		reg = <0x1>;
+	};
+	phy2: ethernet-phy@2 {
+		reg = <0x2>;
+	};
+	phy5: ethernet-phy@5 {
+		reg = <0x5>;
+	};
+};
+
+&gswip_ports {
+	port@4 {
+		//compatible = "lantiq,xrx200-pdi-port";
+		reg = <4>;
+		phy-mode = "internal";
+		phy-handle = <&phy4>;
+		label = "lan2";
+	};
+	port@1 {
+		//compatible = "lantiq,xrx200-pdi-port";
+		reg = <1>;
+		phy-mode = "internal";
+		phy-handle = <&phy1>;
+		label = "lan1";
+	};
+	port@2 {
+		//compatible = "lantiq,xrx200-pdi-port";
+		reg = <2>;
+		phy-mode = "internal";
+		phy-handle = <&phy2>;
+		label = "lan3";
+	};
+	port@5 {
+		//compatible = "lantiq,xrx200-pdi-port";
+		reg = <5>;
+		phy-mode = "rgmii";
+		phy-handle = <&phy5>;
+		label = "lan4";
+	};
+
+};
+
+&stp {
+	status = "okay";
+	lantiq,phy1 = <0x2>;
+	lantiq,phy2 = <0x2>;
+	lantiq,phy3 = <0x2>;
+	lantiq,phy5 = <0x2>;
```

Comment:
Drop lantiq,phy5 = <0x2>;
on the xrx300 there are only three internal PHY which can control some bits of cascade automaticaly.